ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -411,24 +411,36 @@ private IndexValue findKey(StoreKey key, FileSpan fileSpan, IndexEntryType type)
                 true).descendingMap();
         metrics.segmentSizeForExists.update(segmentsMapToSearch.size());
       }
+      int segmentsSearched = 0;
       for (Map.Entry<Offset, IndexSegment> entry : segmentsMapToSearch.entrySet()) {
+        segmentsSearched++;
         logger.trace("Index : {} searching index with start offset {}", dataDir, entry.getKey());
         IndexValue value = entry.getValue().find(key);
         if (value != null) {
           logger.trace("Index : {} found value offset {} size {} ttl {}", dataDir, value.getOffset(), value.getSize(),
               value.getExpiresAtMs());
           if (type.equals(IndexEntryType.ANY)) {
             retValue = value;
+            if(!entry.getValue().isMapped()){
+              metrics.blobFoundInActiveSegmentCount.inc();
+            }
             break;
           } else if (type.equals(IndexEntryType.DELETE) && value.isFlagSet(IndexValue.Flags.Delete_Index)) {
             retValue = value;
+            if(!entry.getValue().isMapped()){
+              metrics.blobFoundInActiveSegmentCount.inc();
+            }
             break;
           } else if (type.equals(IndexEntryType.PUT) && !value.isFlagSet(IndexValue.Flags.Delete_Index)) {
             retValue = value;
+            if(!entry.getValue().isMapped()){
+              metrics.blobFoundInActiveSegmentCount.inc();
+            }
             break;
           }
         }
       }
+      metrics.segmentsAccessedPerBlobCount.inc(segmentsSearched);
```

Comment:
Could you explain how this is useful, especially as a counter?

I think you might have wanted `Histogram` behavior but I think this is information that can be derived quite easily. 

Could you also give an example of how this metric is useful? i.e how would you use this information?